framework/src/modules/interoperability/base_interoperability_module.ts

```diff
@@ -189,28 +181,19 @@ export abstract class BaseInteroperabilityModule extends BaseInteroperableModule
 		}
 	}
 
-	protected _verifyTerminatedStateAccountsCommon(
-		terminatedStateAccounts: TerminatedStateAccountWithChainID[],
-		mainchainID: Buffer,
-	) {
+	protected _verifyTerminatedStateAccountsIDs(chainIDs: Buffer[], mainchainID: Buffer) {
 		// Each entry stateAccount in terminatedStateAccounts has a unique stateAccount.chainID
-		const chainIDs = terminatedStateAccounts.map(a => a.chainID);
 		if (!objectUtils.bufferArrayUniqueItems(chainIDs)) {
 			throw new Error(`terminatedStateAccounts don't hold unique chainID.`);
 		}
 
 		// terminatedStateAccounts is ordered lexicographically by stateAccount.chainID
-		const sortedByChainID = [...terminatedStateAccounts].sort((a, b) =>
-			a.chainID.compare(b.chainID),
-		);
-
-		for (let i = 0; i < terminatedStateAccounts.length; i += 1) {
-			const stateAccountWithChainID = terminatedStateAccounts[i];
-			if (!stateAccountWithChainID.chainID.equals(sortedByChainID[i].chainID)) {
-				throw new Error('terminatedStateAccounts must be ordered lexicographically by chainID.');
-			}
+		if (!objectUtils.isBufferArrayOrdered(chainIDs)) {
+			throw new Error('terminatedStateAccounts must be ordered lexicographically by chainID.');
+		}
 
-			this._verifyChainID(stateAccountWithChainID.chainID, mainchainID, 'stateAccount.');
+		for (const chainID of chainIDs) {
```

Comment:
Minor improvement comment: This could also be removed from here and call `this._verifyChainID(chainID, mainchainID, 'stateAccount.')` inside the loop of `verifyTerminatedStateAccounts` (e.g., line 366 in mainchain/module.ts and line 329 in sidechain/module.ts), since we are already running for-loops there..